src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3703,12 +3703,17 @@ def list_vmss_instance_public_ips(cmd, resource_group_name, vm_scale_set_name):
     return [r for r in result if r.ip_address]
 
 
-def reimage_vmss(cmd, resource_group_name, vm_scale_set_name, instance_id=None, no_wait=False):
+def reimage_vmss(cmd, resource_group_name, vm_scale_set_name, instance_id=None, instance_ids=None, no_wait=False):
     client = _compute_client_factory(cmd.cli_ctx)
-    if instance_id:
-        for instance in instance_id:
-            sdk_no_wait(no_wait, client.virtual_machine_scale_set_vms.begin_reimage,
-                        resource_group_name, vm_scale_set_name, instance)
+    if instance_id and not instance_ids:
+        instance_ids = instance_id
+
+    if instance_ids:
+        VirtualMachineScaleSetVMInstanceIDs = cmd.get_models('VirtualMachineScaleSetVMInstanceIDs')
+        instance_ids = VirtualMachineScaleSetVMInstanceIDs(instance_ids=instance_ids)
```

Comment:
Better to change the name if its type changes, in order to make the code easier to read.

Reply:
Good suggestion! Updated

Reply:
Use `instance_ids` as variable name just to keep consistent with the generally existing code style, such as
https://github.com/Azure/azure-cli/blob/917fba1067dc6ebcf90b6e0a36950e40c15a5d97/src/azure-cli/azure/cli/command_modules/vm/custom.py#L3541-L3558

Reply:
Ugh, `vm_instance_i_ds` is certainly ugly.